### PR TITLE
feat(oci): publish helm chart to xpkg.upbound.io

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
Publish Helm Chart in Addition to `xpkg.upbound.io/upbound/universal-crossplane`
remove unused variable `BOOTSTRAPPER_TAG`

OCI Flow:
During the publish target, the chart will be made available in `xpkg.upbound.io/upbound/universal-crossplane`

Examples:

branch: main
```
xpkg.upbound.io/upbound/universal-crossplane:1.17.0-up.1.rc.0.2.g095726e
```

branch: release-1.17
```
xpkg.upbound.io/upbound/universal-crossplane:1.17.0-up.1
```

we will remove the publish in charts.upbound.io in a later release!

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested
we use the same flow with mcp-connector

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,9 +25,6 @@ CROSSPLANE_REPO := https://github.com/upbound/crossplane.git
 CROSSPLANE_TAG := v1.16.0-up.1
 CROSSPLANE_COMMIT := v1.16.0-up.1
 
-BOOTSTRAPPER_TAG := $(VERSION)
```
Comment (haarchri) [inline Makefile:28]: this variable is unused

File: makelib/helmoci.mk [hunk 2]
```diff
@@ -0,0 +1,108 @@
+# Copyright 2023 The Upbound Authors. All rights reserved.
```
Comment (tnthornton) [inline makelib/helmoci.mk:1]: This is the second or 3rd place I've seen this copied around to recently. Would it make sense to move this the build submodule?

**Review by tnthornton — APPROVED** [review]

File: makelib/helmoci.mk (hunk 2 above)
Comment (haarchri) [reply makelib/helmoci.mk:1]: we deprecated the submodule - so i wonder what is the way to go

Comment (tnthornton) [reply makelib/helmoci.mk:1]: I thought we deprecated the build submodule, but we have https://github.com/crossplane/build for OSS repos 🤷‍♂️

Comment (haarchri) [reply makelib/helmoci.mk:1]: okay will open a PR in crossplane/build and will followup

**Pull request merged**